packages/xstate-lit/test/UseActor.ts

```diff
@@ -0,0 +1,71 @@
+import { html, LitElement } from 'lit';
+import type { Snapshot } from 'xstate';
+import { fromPromise } from 'xstate/actors';
+import { fetchMachine } from './fetchMachine.ts';
+import { UseMachine } from '../src/index.ts';
+
+const onFetch = () =>
+  new Promise<string>((res) => {
+    setTimeout(() => res('some data'), 50);
+  });
+
+const fMachine = fetchMachine.provide({
+  actors: {
+    fetchData: fromPromise(onFetch)
+  }
+});
+
+export class UseActor extends LitElement {
+  fetchController: UseMachine<typeof fetchMachine> = new UseMachine(this, {
+    machine: fMachine,
+    options: {
+      snapshot: this.persistedState
+    }
+  });
+
+  get persistedState(): Snapshot<any> | undefined {
+    return undefined;
+  }
+
+  override createRenderRoot() {
+    return this;
+  }
+
+  override render() {
+    return html`
+      <slot></slot>
+      <div>
+        ${this.fetchController.snapshot.matches('idle')
+          ? html`
+              <button
+                @click=${() => this.fetchController.send({ type: 'FETCH' })}
```

Comment:
This may be a super small DX improvement, but the controller could offer a helper that created an event handler for the user, instead of requiring an arrow function. It would also eliminate the closure creation on every render for a small perf gain.

Something like
```ts
html`
    <button
      @click=${this.fetchController.sendHandler({ type: 'FETCH' })}
```

`sendHandler()` is probably a terrible name, but I couldn't think of something better right now.

Reply:
I’m not quite sure what you're suggesting here. Could you please clarify?

Here is a repository where I created some demos before making the PR. If possible, you can add your suggestions there:

https://github.com/oscarmarina/XstateController/blob/main/xstate-lit/src/UseMachine.ts

Reply:
So right now you need to use an arrow function in the click handler:

```ts
html`
  <button
    @click=${() => this.fetchController.send({ type: 'FETCH' })}
  >
    Fetch
  </button>
`
```

This is fine, but it does allocate a new function object for this binding every render, and it's a slight bit of boilerplate.

Instead you can have the controller allocate a function object once and reuse it every render.

Using it would look like:

```ts
html`
  <button
    @click=${this.fetchController.sendHandler({ type: 'FETCH' })}
  >
    Fetch
  </button>
`
```

and implementing it would look like:

```ts
class MachineController {
  #sendHandler;
  
  /** Returns a callback that calls this.send() with the given arguments **/
  sendHandler(...args) {
    return this.#sendHandler ??= () => this.send(...args);
  }
}
```